File: Java/01.Java-Basic/08.WhileLoop/src/whileLoop/exercise/OldBooks.java
package whileLoop.exercise;

import java.util.Scanner;

public class OldBooks {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int counter = 0;
        Boolean isFound = false;
        String myBook = scanner.nextLine();
        String input = scanner.nextLine();
        while (!input.equals("No More Books")) {
            if (input.equals(myBook)) {
                isFound = true;
                break;
            }

            counter++;
            input = scanner.nextLine();
        }

        if (isFound) {
            System.out.printf("You checked %d books and found it.\n", counter);
        } else {
            System.out.printf("The book you search is not here!\n");
            System.out.printf("You checked %d books.", counter);
        }
    }
}

File: JavaScript/01.JavaScriptBasic/10While-Loop-Exercise/src/old-books.js
function check(params) {
    let favority = params[0];
    let index = 1;
    let data = params[index];
    let isFound = false;
    let booksCounter = 0;
    while (data !== "No More Books") {
        if (data === favority) {
            isFound = true;
            break;
        }

        booksCounter++;
        index++;
        data = params[index];
    }

    if (isFound) {
        console.log(`You checked ${booksCounter} books and found it.`);
    } else {
        console.log(`The book you search is not here!`);
        console.log(`You checked ${booksCounter} books.`);
    }
}

check(["The Spot",
"Hunger Games",
"Harry Potter",
"Torronto",
"Spotify",
"No More Books"]);

check(["Troy",
"Stronger",
"Life Style",
"Troy"]);